src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -25,13 +26,33 @@ internal class DefaultRazorLanguageServerCustomMessageTarget : RazorLanguageServ
         private readonly JoinableTaskFactory _joinableTaskFactory;
         private readonly LSPRequestInvoker _requestInvoker;
         private readonly RazorUIContextManager _uIContextManager;
+        private readonly IDisposable _razorReadyListener;
+
+        private const string RazorReadyFeature = "Razor-Initialization";
 
         [ImportingConstructor]
         public DefaultRazorLanguageServerCustomMessageTarget(
             LSPDocumentManager documentManager,
             JoinableTaskContext joinableTaskContext,
             LSPRequestInvoker requestInvoker,
-            RazorUIContextManager uIContextManager)
+            RazorUIContextManager uIContextManager,
+            IRazorAsynchronousOperationListenerProviderAccessor asyncOpListenerProvider) :
+                this(
+                    documentManager,
+                    joinableTaskContext,
+                    requestInvoker,
+                    uIContextManager,
+                    asyncOpListenerProvider.GetListener(RazorReadyFeature).BeginAsyncOperation(RazorReadyFeature))
+        {
+        }
+
+        // Testing constructor
+        internal DefaultRazorLanguageServerCustomMessageTarget(
+            LSPDocumentManager documentManager,
+            JoinableTaskContext joinableTaskContext,
+            LSPRequestInvoker requestInvoker,
+            RazorUIContextManager uIContextManager,
+            IDisposable razorReadyListener)
```

Comment:
Doing this because `IRazorAsynchronousOperationListenerProviderAccessor` is internal and Microsoft.VisualStudio.LanguageServerClient.Razor.Test does not have IVT to it.